test/filters/test_crop_hull.cpp

```diff
@@ -0,0 +1,382 @@
+      insideCloud->push_back(insidePointGenerator());
+      outsideCloud->push_back(outsidePointGenerator());
+      if (randomBool()) {
+        insideIndicesForMixedCloud.push_back(i);
+        mixedCloud->push_back(insidePointGenerator());
+      }
+      else {
+        mixedCloud->push_back(outsidePointGenerator());
+      }
+    }
+    testData.emplace_back(insideIndicesForInsideCloud, insideCloud);
+    testData.emplace_back(insideIndicesForOutsideCloud, outsideCloud);
+    testData.emplace_back(insideIndicesForMixedCloud, mixedCloud);
+    return testData;
+  }
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST (PCL, ConvexHull_2dsquare)
+{
+  pcl::PointCloud<pcl::PointXYZ> baseOffsetList;
+  baseOffsetList.emplace_back(5, 1, -3);
+  baseOffsetList.emplace_back(1, 5, -3);
+  for (pcl::PointXYZ const & baseOffset : baseOffsetList)
+  {
+    pcl::ConvexHull<pcl::PointXYZ> convexHull;
+    {
+      pcl::PointCloud<pcl::PointXYZ>::Ptr inputCloud (new pcl::PointCloud<pcl::PointXYZ> ());
+      inputCloud->emplace_back(0.0f, 0.0f, 0.0f);
+      inputCloud->emplace_back(0.0f, 1.0f, 0.0f);
+      inputCloud->emplace_back(1.0f, 0.0f, 0.0f);
+      inputCloud->emplace_back(1.0f, 1.0f, 0.0f);
+      inputCloud->emplace_back(0.0f, 0.0f, 0.1f);
+      inputCloud->emplace_back(0.0f, 1.0f, 0.1f);
+      inputCloud->emplace_back(1.0f, 0.0f, 0.1f);
+      inputCloud->emplace_back(1.0f, 1.0f, 0.1f);
+      for (pcl::PointXYZ & p : *inputCloud) {
+        p.getVector3fMap() += baseOffset.getVector3fMap();
+      }
+
+      convexHull.setDimension(3);
+      convexHull.setInputCloud(inputCloud);
+    }
+    Checker checker(convexHull, 2);
+
+    std::mt19937 gen(12345u);
+    std::uniform_real_distribution<float> rd (0.0f, 1.0f);
+    auto insidePointGenerator = [&rd, &gen, baseOffset] () {
+      pcl::PointXYZ p(rd(gen), rd(gen), 1.0);
+      p.getVector3fMap() += baseOffset.getVector3fMap();
+      return p;
+    };
+    auto outsidePointGenerator = [&rd, &gen, baseOffset] () {
+      pcl::PointXYZ p(rd(gen) + 2., rd(gen) + 2., rd(gen) + 2.);
+      p.getVector3fMap() += baseOffset.getVector3fMap();
+      return p;
+    };
+    std::vector<std::pair<pcl::Indices, pcl::PointCloud<pcl::PointXYZ>::ConstPtr>> testData =
+      createTestData(insidePointGenerator, outsidePointGenerator);
+
+    for (auto const & entry : testData)
+    {
+      checker.check(entry.first, entry.second);
+    }
+  }
+}
+
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+TEST (PCL, issue_1657_CropHull3d_not_cropping_inside)
+{
+  typedef pcl::PointCloud<pcl::PointXYZ> PointCloud;
+
+  pcl::CropHull<pcl::PointXYZ> cropHullFilter(true);
+  PointCloud::Ptr hullCloud(new PointCloud());
+  PointCloud::Ptr hullPoints(new PointCloud());
+  std::vector<pcl::Vertices> hullPolygons;
+
+  hullCloud->clear();
+  {
```

Comment:
Please employ loops here. `for (int i = 0, sign = -1, i < 2; sign *= -1, ++i)` is a simple alternative. Nest 3 times to replace L242-L277